csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DependencyManager.cs

```diff
@@ -172,10 +180,38 @@ private void RemoveUnnecessaryNugetPackages(bool existsNetCoreRefNugetPackage, b
                 RemoveNugetPackageReference("microsoft.aspnetcore.app.ref");
             }
 
-            if (existsNetCoreRefNugetPackage && existsNetFrameworkRefNugetPackage)
+            // Multiple dotnet framework packages could be present. We keep only one.
+            // The order of the packages is important, we're keeping the first one that is present in the nuget cache.
+            var packagesInPrioOrder = new (bool isPresent, string prefix)[]
             {
-                // Multiple packages are available, we keep only one:
-                RemoveNugetPackageReference("microsoft.netframework.referenceassemblies.");
+                // net7.0, ... net5.0, netcoreapp3.1, netcoreapp3.0
+                (existsNetCoreRefNugetPackage, "microsoft.netcore.app.ref"),
+                // net48, ..., net20
+                (existsNetFrameworkRefNugetPackage, "microsoft.netframework.referenceassemblies."),
+                // netstandard2.1
+                (existsNetstandardLibRefNugetPackage, "netstandard.library.ref"),
+                // netstandard2.0
+                (existsNetstandardLibNugetPackage, "netstandard.library")
+            };
+
+            for (var i = 0; i < packagesInPrioOrder.Length; i++)
```

Comment:
Maybe consider to replace the loop with something like
```csharp
packagesInPrioOrder
                .SkipWhile(package => !package.isPresent)
                .Skip(1) // Keep the first package with the highest priority
                .Where(package => package.isPresent)
                .ForEach(package => RemoveNugetPackageReference(package.prefix)); // Remove all the lower priority packages.
```
The outer loop basically finds the first element where the package is present and the first time we reach the end of the loop we break out.